### PR TITLE
[WIP] Orca: extend bigdl-submit CLI.

### PR DESCRIPTION
**Pull request by lalalapotter**
## Description

This PR is to extend bigdl-submit CLI.

## User Usage

For the previous version, we only support users (which cannot install conda) to run programs as below:
```python
1. Download BigDL assembly package.
2. Run spark-submit command:
    spark-submit \
        --master yarn \
        --deploy-mode client \
        --archives /path/to/environment.tar.gz#environment \
        --py-files /path/to/python.zip \
        --jars /path/to/jars \
        train.py
```

For currently, we could do this work like:
```python
bigdl-submit
    --master yarn \
    --deploy-mode client \
    --archives /path/to/environment.tar.gz#environment \
    train.py
```

This new extending `bigdl-submit` CLI could provide users an automatically way to run their programs with none-conda directly, what they need to do is just prepare the conda archive.

### CODE REVIEW

**Comment by lalalapotter** [comment]
Need to add tests.

**Comment by hkvision** [comment]
Please add the current command how users would use here

File: scripts/bigdl-submit [hunk 1]
```diff
@@ -2,21 +2,64 @@
 
 set -e
 
-#detect paths
+# usage
+usage() { echo "Usage: $0 [-a|--archives <string#string> -h|--help]" 1>&2; exit 1; }
```
Comment (hkvision) [inline scripts/bigdl-submit:6]: will this --archives conflict with spark-submit's archive?

Comment (lalalapotter) [reply scripts/bigdl-submit:6]: No, I have store the original spark arguments in `cmd`.

File: scripts/bigdl-submit [hunk 2]
```diff
@@ -2,21 +2,64 @@
...
+
+cmd=$@
+# get options
+set +e
+ARGS=`getopt -q -o a:h --long archives:,help -- "$@"`
+set -e
+eval set -- "$ARGS"
+
+# extract options and their arguments into variables.
+archives="./environment.tar.gz"
```
Comment (sgwhat) [inline scripts/bigdl-submit:16]: Can users name the conda archive randomly?

File: scripts/bigdl-submit [hunk 3]
```diff
@@ -2,21 +2,64 @@
...
 set -e
 
-#detect paths
+# usage
+usage() { echo "Usage: $0 [-a|--archives <string#string> -h|--help]" 1>&2; exit 1; }
+
+cmd=$@
+# get options
+set +e
+ARGS=`getopt -q -o a:h --long archives:,help -- "$@"`
```
Comment (sgwhat) [inline scripts/bigdl-submit:11]: Users could ignore `--` after a option now?

Comment (lalalapotter) [reply scripts/bigdl-submit:11]: Yes, I have added break after archives.

File: scripts/bigdl-submit [hunk 4]
```diff
@@ -2,21 +2,145 @@
...
+            then
+                dir=${conf_map["spark.pyspark.python"]}
+            fi
+        fi
+    else
+        if [ $deploy_mode == "cluster" ]
+        then
+            if [ conf_map["spark.yarn.appMasterEnv.PYSPARK_PYTHON"]]
+                dir=${conf_map["spark.yarn.appMasterEnv.PYSPARK_PYTHON"]}
+            fi
```
Comment (sgwhat) [inline scripts/bigdl-submit:89]: Guess we may need `spark.executorEnv.PYSPARK_PYTHON` here?

**Comment by lalalapotter** [comment]
According to customer feedback, their server does not support the use of source to activate the environment, maybe we need to consider other implementation methods.

**Pull request closed without merging**